### PR TITLE
Remove old promote step

### PR DESCRIPTION
**Pull request by logand22**
This PR removes a step from drone that is now failing in v11 promotion events. See https://drone.platform.teleport.sh/gravitational/teleport/16096/5/5 for the failing drone example. 

It also includes a change to `mac.go` in dronegen that was added to dronegen but not updated through a call to `make dronegen`.  See https://github.com/gravitational/teleport/pull/15874

Credit to @wadells for finding the issue. 

## Testing
https://drone.platform.teleport.sh/gravitational/teleport/16121/5/1

### CODE REVIEW

**Review by wadells — APPROVED** [review]

**Comment by zmb3** [comment]
@Joerger are we okay with the `LIBPCSLITE` changes here?

File: .drone.yml [hunk 1]
```diff
@@ -576,7 +576,7 @@ steps:
   - cd $WORKSPACE_DIR/go/src/github.com/gravitational/teleport
   - build.assets/build-fido2-macos.sh build
   - export PKG_CONFIG_PATH="$(build.assets/build-fido2-macos.sh pkg_config_path)"
-  - make clean release OS=$OS ARCH=$ARCH FIDO2=yes TOUCHID=yes LIBPCSCLITE=yes
```
Comment (r0mant) [inline .drone.yml:579]: Why was `LIBPCSCLITE=yes` removed here and below? cc @jakule

**Review by r0mant — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 1 above)
Comment (logand22) [reply .drone.yml:579]: Please see https://github.com/gravitational/teleport/pull/15874 where this value was updated in `dronegen/mac.go` but was not updated in the `.drone.yml` file via `make dronegen`.

Comment (wadells) [reply .drone.yml:579]: @Joerger added this change here: https://github.com/gravitational/teleport/pull/15874/files#diff-5a952c8230e4b996fed9112933bb065ab4efbf132bf59ef83811e8c5860cdd51

But he forgot to run dronegen afterwards.

Comment (r0mant) [reply .drone.yml:579]: Gotcha - @jakule @Joerger can you confirm this is intended change?

Comment (Joerger) [reply .drone.yml:579]: Yes this is correct, sorry for the confusion.

**Review by Joerger — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**